openinfra_core/src/main/java/de/btu/openinfra/backend/db/rbac/OpenInfraRealm.java

```diff
@@ -0,0 +1,98 @@
+package de.btu.openinfra.backend.db.rbac;
+
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.shiro.authc.AuthenticationException;
+import org.apache.shiro.authc.AuthenticationToken;
+import org.apache.shiro.authc.SaltedAuthenticationInfo;
+import org.apache.shiro.authc.SimpleAuthenticationInfo;
+import org.apache.shiro.authc.UsernamePasswordToken;
+import org.apache.shiro.authz.AuthorizationInfo;
+import org.apache.shiro.authz.SimpleAuthorizationInfo;
+import org.apache.shiro.crypto.hash.Sha256Hash;
+import org.apache.shiro.realm.AuthorizingRealm;
+import org.apache.shiro.subject.PrincipalCollection;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+import org.apache.shiro.util.SimpleByteSource;
+
+import de.btu.openinfra.backend.db.daos.rbac.SubjectDao;
+import de.btu.openinfra.backend.db.jpa.model.rbac.RolePermission;
+import de.btu.openinfra.backend.db.jpa.model.rbac.Subject;
+import de.btu.openinfra.backend.db.jpa.model.rbac.SubjectRole;
+
+/**
+ * This is the OpenInfraRealm which is used to retrieve user-specific 
+ * information from database. The idea behind this implementation was: 
+ * 'keep it stupid and simple' and is closely related to the implementation
+ * of the origin jdbcReal.
+ * 
+ * @author <a href="http://www.b-tu.de">BTU</a> DBIS
+ *
+ */
+public class OpenInfraRealm extends AuthorizingRealm {
+	
+	@Override
+	protected SaltedAuthenticationInfo doGetAuthenticationInfo(
+			AuthenticationToken token) throws AuthenticationException {
+		
+		UsernamePasswordToken upt = (UsernamePasswordToken)token;
+		Subject s = new SubjectDao().readModel(upt.getUsername());
+		
+		return new SimpleAuthenticationInfo(
+				new SimplePrincipalCollection(
+						upt.getUsername(),
+						OpenInfraRealmNames.LOGIN.name()), 
+						s.getPassword(),
+						new SimpleByteSource(s.getSalt().toString()));
+	}
+	
+	@Override
+	protected AuthorizationInfo doGetAuthorizationInfo(
+			PrincipalCollection principals) {
+		
+		// Retrieve the login from principals (there might be multiple)
+		List<String> login = new LinkedList<String>();
+		for(Object o : principals.fromRealm(OpenInfraRealmNames.LOGIN.name())) {
+			login.add(o.toString());
+		}
+		
+        Subject s = new SubjectDao().readModel(login.get(0));

```

Comment:
If their can be multiple logins, why is the first always correct? What about the rest?